### ISSUE TITLE
Model creation sometimes fails with NoOverlapError

### ISSUE THREAD

**Issue by keflavich** [astropy/photutils#1589]
Traceback:
```
Model image:   0%|          | 4/201472 [00:00<2:32:54, 21.96it/s]
Traceback (most recent call last):
  File "/blue/adamginsburg/adamginsburg/jwst/brick/analysis/crowdsource_catalogs_long.py", line 726, in <module>
    main()
  File "/blue/adamginsburg/adamginsburg/jwst/brick/analysis/crowdsource_catalogs_long.py", line 699, in main
    residual = phot_.make_residual_image(data, (11, 11))
  File "/blue/adamginsburg/adamginsburg/repos/photutils/photutils/psf/photometry.py", line 1364, in make_residual_image
    residual = data - self.make_model_image(data.shape, psf_shape)
  File "/blue/adamginsburg/adamginsburg/repos/photutils/photutils/psf/photometry.py", line 1328, in make_model_image
    slc_lg, _ = overlap_slices(shape, psf_shape, (y0, x0), mode='trim')
  File "/blue/adamginsburg/adamginsburg/repos/astropy/astropy/nddata/utils.py", line 122, in overlap_slices
    raise NoOverlapError("Arrays do not overlap.")
astropy.nddata.utils.NoOverlapError: Arrays do not overlap.
```

Code (not MWE):
```python
                phot_ = IterativePSFPhotometry(finder=daofind_tuned,
                                               localbkg_estimator=LocalBackground(5, 15),
                                               psf_model=dao_psf_model,
                                               fitter=LevMarLSQFitter(),
                                               maxiters=2,
                                               fit_shape=(5, 5),
                                               aperture_radius=2*fwhm_pix,
                                               progress_bar=True
                                              )
                result2 = phot_(data)
                residual = phot_.make_residual_image(data, (11, 11))
```

This results from a bad fit:
```
<Table length=201472>
      x_fit              y_fit              x_init             y_init
     float64            float64            float64            float64
------------------ ------------------ ------------------ ------------------
 4638.255588025306  15.71238415793701   4638.42933208721 15.738809631503454
 4183.458422565426  19.97167175541903  4183.326348304083  19.88849130623425
  4194.92753151351 19.902057977130056  4194.858499418677  19.88066861559819
 4557.967301274507 19.868096971316845  4557.976122441039 19.795241774157056
 4398.316580378928 -97.12638709537387  4282.069407322359  20.46607216037275
```
note `y_fit=-97`

I think the `overlap_slices` call should be wrapped with a try/except and warn, instead of raise, for an out-of-bounds input.

Also it would be nice to provide a utility function to filter out fitted sources that are >1 PSF (or >1 footprint) from the init.  I'll do that by hand, but this is clearly not a fit.

**Comment by keflavich** [comment]
OK... wait... it already does that.  How am I getting this exception?
https://github.com/astropy/photutils/blob/d2d7a98709ec241a79216f626d5ab7c24869b55e/photutils/psf/photometry.py#L998-L1002

**Comment by keflavich** [comment]
Ah, the problem is in `IterativePSFPhotometry`; `PSFPhotometry` already has this behavior

**Comment by keflavich** [comment]
Closed by #1590

**Comment by SterlingYM** [comment]
The same isssue happens in ```PSFPhotometry._define_fit_data()``` when running the code below. I can see the code is written so that it explicitly raises error, rather than ignoring it. 
I can see that this is also due to the poor fit. Is this something that can be ignored, and if so, can this be changed to just warnings?

MWE:
~~~python
from photutils.datasets import make_test_psf_data, make_noise_image
from photutils.detection import DAOStarFinder
from photutils.psf import IntegratedGaussianPRF, IterativePSFPhotometry, SourceGrouper

### try the code below with nsources=100 and 300
# nsources = 100 # succeeds
nsources = 300 # fails with NoOverlapError

# define psf
sigma_psf = 1.5
psf_model = IntegratedGaussianPRF(flux=1, sigma=sigma_psf)

# generate test data
shape = (150,150)
psf_shape = (11,11)
data, true_params = make_test_psf_data(shape, psf_model, psf_shape,
                                       nsources=nsources, flux_range=(50, 100),
                                       min_separation=1,seed=0)
noise = make_noise_image(data.shape, mean=0, stddev=0.01, seed=0)
data += noise
error = np.abs(noise)

# tools
daofinder = DAOStarFinder(threshold=0.5, fwhm=sigma_psf)
grouper = SourceGrouper(min_separation=3.0*sigma_psf) 

# run photometry
psf_iter = IterativePSFPhotometry(psf_model, fit_shape=(5,5), finder=daofinder,
                                mode='all',grouper=grouper,
                                aperture_radius=3)
phot_result = psf_iter(data, error=error)
~~~
